src/Files.App.Controls/ThemedIcon/ThemedIcon.cs

```diff
@@ -0,0 +1,387 @@
+        }
+
+        private void OutlineIconPathUpdate()
+        {
+            // Updates Outline Icon from Path Data
+
+            if (GetTemplateChild(OutlinePathIconViewBox) is not Viewbox outlineViewBox)
+                return;
+
+            SetPathData(OutlineIconPath, OutlineIconData ?? string.Empty, outlineViewBox);
+        }
+
+        private void LayeredIconContentUpdate()
+        {
+            // Updates Layered Icon from it's Layers
+
+            if (GetTemplateChild(LayeredPathIconViewBox) is not Viewbox layeredViewBox ||
+                GetTemplateChild(LayeredPathCanvas) is not Canvas canvas ||
+                Layers is not ICollection<ThemedIconLayer> layers)
+                return;
+
+            foreach (var layer in layers)
+            {
+                canvas.Children.Add(
+                    new ThemedIconLayer()
+                    {
+                        LayerType = layer.LayerType,
+                        IconColorType = layer.IconColorType,
+                        PathData = layer.PathData,
+                        Opacity = layer.Opacity,
+                        LayerColor = this.Color,
+                        Foreground = this.Foreground,
+                        HorizontalAlignment = HorizontalAlignment.Stretch,
+                        VerticalAlignment = VerticalAlignment.Stretch,
+                        Width = Width,
+                        Height = Height
+                    });
+            }
+        }
+
+        private void SetPathData(string partName, string pathData, FrameworkElement element)
+        {
+            // Updates PathData
+
+            if (string.IsNullOrEmpty(pathData))
+                return;
+
+            var geometry = (Geometry)XamlReader.Load(
+                $"<Geometry xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'>{pathData}</Geometry>");
+
+            if (GetTemplateChild(partName) is Path path)
+            {
+                path.Data = geometry;
+                path.Width = element.Width;
+                path.Height = element.Height;
+            }
+        }
+
+        private void FindOwnerControlStates()
+        {
+            // Finds the owner Control and it's Checked and Enabled state
+
+            //
+            // Check if owner Control is a ToggleButton
+            // Hooks onto Event handlers when IsChecked and IsUnchecked runs
+            // Runs the ToggleChanged event, to set initial value, if the ToggleButton's isChecked is true
+            //
+            // Check if owner Control is an AppBarToggleButton
+            // Hooks onto Event handlers when IsChecked and IsUnchecked runs
+            // Runs the ToggleChanged event, to set initial value, if the AppBarToggleButton's isChecked is true
+            //
+            // Gets the owner Control
+            // Hooks onto Event handlers when IsEnabledChanged runs
+            // Runs the EnabledChanged event to set initial value
+            //
```

Comment:
I thought it may be useful if others come to maintain the control in the future.  It also helped me keep track of the flow as I was writing the code.

I am still a beginner with C#

Reply:
I have put the longer comments into collapsable comment blocks